src/SDKs/Azure.Base/data-plane/Azure.Base/Http/Pipeline/FixedPolicy.cs

```diff
@@ -2,30 +2,47 @@
 // Licensed under the MIT License.
 
 using System;
+using System.Linq;
 
 namespace Azure.Base.Http.Pipeline
 {
-    class FixedPolicy : RetryPolicy {
-        int _maxRetries;
-        TimeSpan _delay;
-        int[] _retriableCodes;
+    internal class FixedPolicy : RetryPolicy
+    {
+        private readonly int _maxRetries;
+
+        private readonly TimeSpan _delay;
+
+        private readonly int[] _retriableCodes;
 
         public FixedPolicy(int[] retriableCodes, int maxRetries, TimeSpan delay)
         {
-            if (retriableCodes == null) throw new ArgumentNullException(nameof(retriableCodes));
+            if (retriableCodes == null)
+            {
+                throw new ArgumentNullException(nameof(retriableCodes));
+            }
 
             _maxRetries = maxRetries;
             _delay = delay;
-            _retriableCodes = retriableCodes;
+
+            _retriableCodes = retriableCodes.ToArray();
```

Comment:
Is this really needed? Do you expect people to make the mistake (of mutating the array after they pass it to us) in really world? Even if they do, what's the worst that can happen? Lastly, if we let people mutate the policy settings on the options instance, the list will be mutable anyway.

Reply:
It's just being a well-behaving API and avoid mutating things we're not supposed to.

> Even if they do, what's the worst that can happen? 

Binary search won't work and requests would not be retried when they should.